plugins/module_utils/network/sonic/config/ospfv2/ospfv2.py

```diff
@@ -0,0 +1,773 @@
+        requests = []
+
+        for cmd in commands:
+            vrf_name = cmd.get('vrf_name')
+            sub_commands = deepcopy(cmd)
+            sub_commands.pop('vrf_name')
+            payload = {}
+            self._get_create_request_path_from_dict(sub_commands, ospf_attributes, payload)
+            if payload:
+                requests.append({'path': ospf_path.format(vrf_name=vrf_name), 'method': PATCH, 'data': {'openconfig-network-instance:ospfv2': payload}})
+        return requests
+
+    def _get_create_request_path_from_dict(self, command, ospf_dict, payload):
+        for attr in ospf_dict:
+            if attr in command and command[attr] is not None:
+                if not isinstance(ospf_dict[attr], dict):
+                    path = ospf_dict[attr]
+                    request_body = path.split('/')
+                    last_body = request_body[-1]
+                    if '=' in last_body:
+                        last_body = last_body.split('=')[0]
+                    payload_iter = payload
+                    for body in request_body[1:-1]:
+                        payload_iter.setdefault(body, {})
+                        payload_iter = payload_iter[body]
+                    if attr == 'advertise_router_id':
+                        for ip in command['advertise_router_id']:
+                            payload_iter.setdefault('helper', [])
+                            payload_iter['helper'].append({'neighbour-id': ip, 'config': {'neighbour-id': ip}})
+                    elif attr == 'abr_type':
+                        payload_iter[last_body] = 'openconfig-ospfv2-ext:' + command[attr].upper()
+                    elif attr == 'log_adjacency_changes':
+                        payload_iter[last_body] = 'openconfig-ospfv2-ext:' + command[attr].upper()
+                    else:
+                        if attr == 'grace_period':
+                            command['enable'] = True
+                            payload_iter['enabled'] = True
+                        payload_iter[last_body] = command[attr]
+                elif attr == 'redistribute':
+                    self._get_create_redistribute_request(command[attr], ospf_dict[attr], attr, payload)
+                elif attr == 'passive_interfaces' or attr == 'non_passive_interfaces':
+                    self._get_create_passive_interfaces_request(command[attr], attr, payload)
+                else:
+                    self._get_create_request_path_from_dict(command[attr], ospf_dict[attr], payload)
+
+    def _get_create_redistribute_request(self, command, ospf_dict, attr, payload):
+        protocol = 'DEFAULT_ROUTE'
+        path = redistribute_delete_path
+        request_body = path.split('/')
+        payload_iter = payload
+        for body in request_body[1:-1]:
+            payload_iter.setdefault(body, {})
+            payload_iter = payload_iter[body]
+        payload_iter.setdefault('distribute-list', [])
```

Comment:
Since the payload format is the same for all redistribute sub-commands, the payload could be initialized statically similar to '_get_create_passive_interfaces_request'.